server/channels/api4/user_local.go

```diff
@@ -323,26 +323,41 @@ func localDeleteUser(c *Context, w http.ResponseWriter, r *http.Request) {
 	auditRec := c.MakeAuditRecord("localDeleteUser", audit.Fail)
 	defer c.LogAuditRec(auditRec)
 
-	user, err := c.App.GetUser(userId)
-	if err != nil {
-		c.Err = err
+	user, appErr := c.App.GetUser(userId)
+	if appErr != nil {
+		c.Err = appErr
 		return
 	}
 	audit.AddEventParameter(auditRec, "user_id", c.Params.UserId)
 	auditRec.AddEventPriorState(user)
 	auditRec.AddEventObjectType("user")
 
+	var job *model.Job
 	if c.Params.Permanent {
-		err = c.App.PermanentDeleteUser(c.AppContext, user)
+		job, appErr = c.App.CreateJob(c.AppContext, &model.Job{
```

Comment:
Do we have other examples of API handlers implicitly starting jobs? I wonder whether this is something the client side should do instead.

Reply:
This is actually a great idea. To keep the API as-is, but change mmctl to invoke a job while deleting the user. That makes no breaking changes, but still allows mmctl to get the job done!